python/example/simple_attack.ipynb

```diff
@@ -0,0 +1,416 @@
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 108,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "person of interest:\n",
+      "\n",
+      "6000.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "person_of_interest = incomes[0]\n",
+    "print('person of interest:\\n\\n{0}'.format(person_of_interest))"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Now consider the case that the attacker knows everything about the data, except for the person of interest's (POI) income, which is considered private.\n",
+    "They can back out the individual's income very easily, just from asking for the mean overall income."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 109,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "poi_income: 6000.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "# attacker information: he already knows everyone else's income, so he certainly compute the follwoing\n",
+    "known_mean = np.mean(incomes[1:])\n",
+    "known_obs = len(incomes) - 1\n",
+    "\n",
+    "# assume the attackers know legitimately the overall mean and number of people in the data...\n",
+    "overall_mean = np.mean(incomes)\n",
+    "n_obs = len(incomes)\n",
+    "\n",
+    "# back out POI's income\n",
+    "poi_income = overall_mean * n_obs - known_obs * known_mean\n",
+    "print('poi_income: {0}'.format(poi_income))"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "The attacker now knows with certainty that the POI has an income of \\$6,000.\n",
+    "\n",
+    "\n",
+    "## Using OpenDP\n",
+    "Let's see what happens if the attacker were made to interact with the data through OpenDP and was given a privacy budget of $\\epsilon = 1$.\n",
+    "We will assume that the attacker is reasonably familiar with differential privacy and believes that they should use tighter data bounds than they know are actually in the data in order to get a less noisy estimate.\n",
+    "They will need to update their `known_mean` accordingly."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 111,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DP mean: 29805.466099517937\n",
+      "Known mean: 26883.913\n"
+     ]
+    }
+   ],
+   "source": [
+    "from opendp.mod import binary_search_chain\n",
+    "from opendp.trans import make_clamp, make_bounded_sum, make_sized_bounded_mean, make_bounded_resize\n",
+    "from opendp.meas import make_base_geometric, make_base_laplace\n",
+    "\n",
+    "enable_features(\"floating-point\")\n",
```

Comment:
Above I enabled "contrib", is it the same as "floating point"?

Reply:
"contrib" encompasses all library components that have not completed the review process. "floating-point" encompasses all measurements that may have floating-point vulnerabilities. You have to enable "floating-point" to use continuous noise samplers like the laplace. We now have mitigations for floating-point attacks in main for the laplace and gaussian samplers, but we'll be slow to drop that flag because floating-point implementations are so prone to DP issues.

Reply:
Not sure which paragraph are you talking about?

Reply:
I think at the time I first reviewed this, you were enabling the same feature twice. It look good now.